idb/postgres/internal/writer/writer.go

```diff
@@ -278,17 +313,33 @@ func getTransactionParticipants(stxnad transactions.SignedTxnWithAD) []basics.Ad
 	return res
 }
 
-func addTransactionParticipation(block *bookkeeping.Block, batch *pgx.Batch) error {
-	for i, stxnad := range block.Payset {
-		// TODO: replace with a function from go-algorand.
-		participants := getTransactionParticipants(stxnad.SignedTxnWithAD)
+func addInnerTransactionParticipation(stxnad transactions.SignedTxnWithAD, round, intra uint64, batch *pgx.Batch) uint64 {
+	finalIntra := intra
```

Comment:
I think there is no reason for a copy.

Reply:
you're technically right, I have trust issues when it comes to arguments so I avoid modifying them.

Reply:
`finalIntra` should be called `next` or similar